### PR TITLE
CPUID Module

### PR DESCRIPTION
**Pull request by gegy1000**
This PR implements a module for interaction with the cpuid instruction. Currently supports listing all available CPU features.

### CODE REVIEW

File: kernel/src/cpuid.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        const DCA = 1 << 50;
+        const SSE4_1 = 1 << 51;
+        const SSE4_2 = 1 << 52;
+        const x2APIC = 1 << 53;
+        const MOVBE = 1 << 54;
+        const POPCNT = 1 << 55;
+        const AES = 1 << 57;
+        const XSAVE = 1 << 58;
+        const OSXSAVE = 1 << 59;
+        const AVX = 1 << 60;
```
Comment (Restioson) [inline kernel/src/cpuid.rs]: ... Docs??

File: kernel/src/cpuid.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
+use core::arch::x86_64::__cpuid;
+
+const CPUID_GET_FEATURES: u32 = 0x1;
+
+bitflags! {
+    pub struct Features: u64 {
+        const FPU = 1 << 0;
+        const VME = 1 << 1;
+        const DE = 1 << 2;
```
Comment (Restioson) [inline kernel/src/cpuid.rs]: Germany?? :P

File: kernel/src/cpuid.rs (hunk 1 above)
Comment (gegy1000) [reply kernel/src/cpuid.rs]: From [OSDev Wiki](https://wiki.osdev.org/CPUID#CPU_Features)... I honestly have no idea.

Comment (Restioson) [reply kernel/src/cpuid.rs]: Maybe look it up and doc

**Comment by Restioson** [comment]
Oops, it is combined. Lol

**Pull request merged**